training/src/anemoi/training/diagnostics/logger.py

```diff
@@ -27,79 +27,46 @@ def get_mlflow_logger(config: BaseSchema) -> None:
-
-    from anemoi.training.diagnostics.mlflow import LOG_MODEL
-    from anemoi.training.diagnostics.mlflow import MAX_PARAMS_LENGTH
-    from anemoi.training.diagnostics.mlflow.logger import AnemoiMLflowLogger
+    # TODO: this seems a bit dirty
+    log_system = logger_config.pop("system")
+    log_terminal = logger_config.pop("terminal")
+    expand_hyperparams = logger_config.pop("expand_hyperparams")
 
+    # Defaults that exist outside the scope of config.diagnostics.log.mlflow
     resumed = config.training.run_id is not None
     forked = config.training.fork_run_id is not None
+    offline = logger_config.get("offline", False)
+
+    # TODO: This does not make sense, and if we get rid of it, the 3 lines above can be removed too
+    if (resumed or forked) and (offline):  # when resuming or forking offline -
+        logging_config["tracking_uri"] = str(save_dir)
 
-    save_dir = config.hardware.paths.logs.mlflow
+    if "save_dir" in logger_config:
+        if logger_config["save_dir"] is None:
+            logger_config["save_dir"] = config.hardware.paths.logs.mlflow
 
-    offline = config.diagnostics.log.mlflow.offline
-    if not offline:
-        tracking_uri = config.diagnostics.log.mlflow.tracking_uri
-        LOGGER.info("AnemoiMLFlow logging to %s", tracking_uri)
-    else:
-        tracking_uri = None
+        # only create save_dir if it's specified in schema (e.g. it's not for Azure)
+        Path(logger_config["save_dir"]).mkdir(parents=True, exist_ok=True)
 
-    if (resumed or forked) and (offline):  # when resuming or forking offline -
-        # tracking_uri = ${hardware.paths.logs.mlflow}
-        tracking_uri = str(save_dir)
-    # create directory if it does not exist
-    Path(config.hardware.paths.logs.mlflow).mkdir(parents=True, exist_ok=True)
-
-    log_hyperparams = True
-    if resumed and not config.diagnostics.log.mlflow.on_resume_create_child:
-        LOGGER.info(
-            (
-                "Resuming run without creating child run - MLFlow logs will not update the"
-                "initial runs hyperparameters with those of the resumed run."
-                "To update the initial run's hyperparameters, set "
-                "`diagnostics.log.mlflow.on_resume_create_child: True`."
-            ),
-        )
-        log_hyperparams = False
-
-    max_params_length = getattr(config.diagnostics.log.mlflow, "max_params_length", MAX_PARAMS_LENGTH)
-    LOGGER.info("Maximum number of params allowed to be logged is: %s", max_params_length)
-    log_model = getattr(config.diagnostics.log.mlflow, "log_model", LOG_MODEL)
-
-    logger = AnemoiMLflowLogger(
-        experiment_name=config.diagnostics.log.mlflow.experiment_name,
-        project_name=config.diagnostics.log.mlflow.project_name,
-        tracking_uri=tracking_uri,
-        save_dir=save_dir,
-        run_name=config.diagnostics.log.mlflow.run_name,
+    logger = instantiate(
+        logger_config,
```

Comment:
are you proposing the pass the logger_config to the init and then handle all the settings there? Otherwise I see a mix at the moment and it's not clear how some options like 'offline' that are removed from this function are later handled in the init

Reply:
Yeah I think that would be cleaner, but this implementation was just a start. I think it would be a bit more straightforward to remove as much of this "preamble" stuff and handle everything in the logger class. I'm a bit new to omegaconf and hydra though so I think my implementation is a bit rough around the edges.